### PR TITLE
pipeline: fixes for installing and testing rook

### LINKED ISSUE #1714: [2] pipeline - fails intermittently att apps(sc) step
**Describe the bug**:
The end-to-end pipeline fails intermittently when installing apps(sc), seemingly when installing harbor.

```bash
STDERR:
  W0811 04:50:00.955416    8479 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "exporter" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "exporter" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "exporter" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "exporter" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  W0811 04:50:00.966526    8479 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (containers "registry", "registryctl" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (containers "registry", "registryctl" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or containers "registry", "registryctl" must set securityContext.runAsNonRoot=true), seccompProfile (pod or containers "registry", "registryctl" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  W0811 04:50:00.978218    8479 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "portal" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "portal" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "portal" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "portal" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  W0811 04:50:01.087999    8479 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "core" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "core" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "core" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "core" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  W0811 04:50:01.297541    8479 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "notary-signer" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "notary-signer" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "notary-signer" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "notary-signer" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  W0811 04:50:01.439284    8479 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "notary-server" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "notary-server" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "notary-server" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "notary-server" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  W0811 04:50:01.454535    8479 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "jobservice" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "jobservice" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "jobservice" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "jobservice" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  W0811 04:50:01.677514    8479 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (containers "data-migrator", "data-permissions-ensurer", "database" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (containers "data-migrator", "data-permissions-ensurer", "database" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or containers "data-migrator", "data-permissions-ensurer", "database" must set securityContext.runAsNonRoot=true), seccompProfile (pod or containers "data-migrator", "data-permissions-ensurer", "database" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  W0811 04:50:01.700873    8479 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "redis" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "redis" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "redis" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "redis" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  W0811 04:50:01.901591    8479 warnings.go:70] would violate PodSecurity "restricted:latest": unrestricted capabilities (container "trivy" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "trivy" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "trivy" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Error: timed out waiting for the condition

NAME                                        READY   STATUS             RESTARTS      AGE
pod/harbor-core-68545858dd-zwn9s            0/1     CrashLoopBackOff   5 (37s ago)   10m
pod/harbor-database-0                       0/1     Pending            0             10m
pod/harbor-exporter-5b98bd765f-zmcc5        0/1     CrashLoopBackOff   5 (75s ago)   10m
pod/harbor-jobservice-976ff67b9-d2tbw       0/1     Pending            0             10m
pod/harbor-notary-server-76965fc6b9-bz7gx   0/1     CrashLoopBackOff   6 (27s ago)   10m
pod/harbor-notary-signer-5c7675d78-cw77v    0/1     CrashLoopBackOff   6 (26s ago)   10m
pod/harbor-portal-56cf4bcd9d-x67x2          1/1     Running            0             10m
pod/harbor-redis-0                          0/1     Pending            0             10m
pod/harbor-registry-b7ff96966-shxss         2/2     Running            0             10m
pod/harbor-trivy-0                          0/1     Pending            0             10m

NAME                           TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
service/harbor-core            ClusterIP   10.233.59.10    <none>        80/TCP,8001/TCP              10m
service/harbor-database        ClusterIP   10.233.27.183   <none>        5432/TCP                     10m
service/harbor-exporter        ClusterIP   10.233.61.240   <none>        8001/TCP                     10m
service/harbor-jobservice      ClusterIP   10.233.50.132   <none>        80/TCP,8001/TCP              10m
service/harbor-notary-server   ClusterIP   10.233.3.245    <none>        4443/TCP                     10m
service/harbor-notary-signer   ClusterIP   10.233.44.107   <none>        7899/TCP                     10m
service/harbor-portal          ClusterIP   10.233.19.11    <none>        80/TCP                       10m
service/harbor-redis           ClusterIP   10.233.12.220   <none>        6379/TCP                     10m
service/harbor-registry        ClusterIP   10.233.56.94    <none>        5000/TCP,8080/TCP,8001/TCP   10m
service/harbor-trivy           ClusterIP   10.233.62.145   <none>        8080/TCP                     10m

NAME                                   READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/harbor-core            0/1     1            0           10m
deployment.apps/harbor-exporter        0/1     1            0           10m
deployment.apps/harbor-jobservice      0/1     1            0           10m
deployment.apps/harbor-notary-server   0/1     1            0           10m
deployment.apps/harbor-notary-signer   0/1     1            0           10m
deployment.apps/harbor-portal          1/1     1            1           10m
deployment.apps/harbor-registry        1/1     1            1           10m
```

**To Reproduce**:

Steps to reproduce the behaviour:

1. Go to 'Actions' on Github
1. Click on 'End-to-end'
1. Click on a failed run
1. See errors

**Expected behaviour**:

End-to-end pipeline doesn't fail when installing apps

**Additional context**:

Try to identify the issue and fix it

### PR DESCRIPTION
**Pull request by lunkan93**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [x] personal data beyond what is necessary for interacting with this pull request
> - [x] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [ ] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [x] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### What does this PR do / why do we need this PR?

Added various fixes after the rook rework and addition of test script.

- Fixes #1714 

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy

### CODE REVIEW

**Comment by linus-astrom** [comment]
If you have made kubespray workable with the pipeline then you should erase this [text](https://github.com/elastisys/compliantkubernetes-apps/blob/fa1ce444b702637519412c07b56e9a0ddf762089/.github/workflows/end-to-end.yml#L58-L59) in this commit aswell.

File: pipeline/config/exoscale/rook/values.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    enabled: true
+    dnsPeers:
+      - cidr: 0.0.0.0/0
+  cluster:
+    preset: "none"
+    mon:
+      resources:
+        requests:
+          cpu: 400m
+          memory: 500Mi
```
Comment (linus-astrom) [inline pipeline/config/exoscale/rook/values.yaml:15]: Question should we have any kind of resource limit on ```mon:``` and ```osd:```or is it fine without?

**Comment by lunkan93** [comment]
> If you have made kubespray workable with the pipeline then you should erase this [text](https://github.com/elastisys/compliantkubernetes-apps/blob/fa1ce444b702637519412c07b56e9a0ddf762089/.github/workflows/end-to-end.yml#L58-L59) in this commit aswell.

Yeah that seems to be some 2 year old text that has just stuck around.. I'll remove it

File: pipeline/config/exoscale/rook/values.yaml (hunk 1 above)
Comment (lunkan93) [reply pipeline/config/exoscale/rook/values.yaml:15]: I don't think we need to override it, the limits do get set to the chart defaults. I lowered the requests in order to make sure that there are enough resources for everything to get scheduled.

**Review by linus-astrom — APPROVED** [review]

**Comment by lunkan93** [comment]
:eyes:

**Review by OlleLarsson — APPROVED** [review]

**Pull request merged**